crates/redeem/src/lib.rs

```diff
@@ -325,21 +325,21 @@ impl<T: Config> Pallet<T> {
             .checked_sub(&fee_wrapped)
             .ok_or(Error::<T>::ArithmeticUnderflow)?;
 
-        // this can overflow for small requested values. As such return AmountBelowDustAmount when this happens
-        let user_to_be_received_btc = vault_to_be_burned_tokens
```

Comment:
I thought this could be expressed more cleanly, which is why I moved it after the `ensure!(..)`